docs/react_native_tutorials/react_native_1_setup_identifiers.md

```diff
@@ -0,0 +1,461 @@
+### Prerequisite configuration
+
+In an ideal world we would install some dependencies, paste some sample code and start our app, but anyone who has
+worked with a non-trivial react-native app probably already knows that we don't currently live in this ideal world.
+Some libraries that we will be depending on make use of the node `crypto` package, or the `crypto.subtle` API of
+browsers. These are not yet available to react-native/expo environments out of the box, so there is a bit of ceremony to
+bundle everything properly.
+
+Also, the metro bundler that react-native uses doesn't yet support the `cjs` file extension (
+see [facebook/metro#535](https://github.com/facebook/metro/issues/535)) which is used by some libraries in the stack, so
+we have to configure these too.
+
+#### `cjs` extension
+
+Please adjust your `metro.config.js` file to look like this:
+
+```js
+// filename: metro.config.js
+const { getDefaultConfig } = require('metro-config')
+const { resolver: defaultResolver } = getDefaultConfig.getDefaultValues()
+
+exports.resolver = {
+  ...defaultResolver,
+  sourceExts: [...defaultResolver.sourceExts, 'cjs'],
+}
+```
+
+#### `crypto` shims
+
+Next, we start setting up the shims that will be required by our libraries.
+
+```bash
+npm i @sinonjs/text-encoding react-native-get-random-values @ethersproject/shims
+npm i -D rn-nodeify babel-plugin-rewrite-require
+```
+
+Next, edit your `package.json` file and add a `postinstall` script:
+
+```json5
+// filename: package.json
+{
+  //...
+  scripts: {
+    // ...
+    postinstall: 'rn-nodeify --install crypto,stream,process --hack',
+  },
+}
+```
+
+Run a `npm install` to trigger all the required to be installed and adjusted and to generate a `./shim.js` file.
+Open `shim.js` and uncomment the last line that says `require('crypto)`.
+
+```js
+// filename: shim.js
+
+// ...
+
+// If using the crypto shim, uncomment the following line to ensure
+// crypto is loaded first, so it can populate global.crypto
+require('crypto')
+```
+
+Now edit your `babel.config.js` file at your project root and add the `babel-plugin-rewrite-require` to it, like so:
+
+```js
+// filename: babel.config.js
+module.exports = function (api) {
+  api.cache(true)
+  return {
+    presets: ['babel-preset-expo'],
+    plugins: [
+      [
+        'babel-plugin-rewrite-require',
+        {
+          aliases: {
+            crypto: 'crypto-browserify',
+            stream: 'stream-browserify',
+          },
+        },
+      ],
+    ],
+  }
+}
+```
+
+Next, we can start adding the shims to the top of our `App.tsx` file. Read
+about [strong random values here](https://docs.ethers.io/v5/cookbook/react-native/#cookbook-reactnative-security).
+
+```tsx
+// filename: App.tsx
+
+/// shims
+import '@sinonjs/text-encoding'
+import 'react-native-get-random-values'
+import './shim.js'
+import '@ethersproject/shims'
+```
+
+Now, install all the pods in your project that came with the new dependencies.
+
+```bash
+npx pod-install
```

Comment:
This command fails on windows (and presumably Linux) with the following error:

```
Scanning for pods...
CocoaPods is only supported on darwin machines
```

Reply:
right; this should be mentioned in the tutorial.
You can't build for iOS on non macOS machines